src/input.rs

```diff
@@ -204,10 +206,18 @@ impl<C: Configuration> Ingredient for IngredientImpl<C> {
         _db: &dyn Database,
         _input: Id,
         _revision: Revision,
-    ) -> MaybeChangedAfter {
+    ) -> VerifyResult {
         // Input ingredients are just a counter, they store no data, they are immortal.
         // Their *fields* are stored in function ingredients elsewhere.
-        MaybeChangedAfter::No(InputAccumulatedValues::Empty)
+        VerifyResult::unchanged()
+    }
+
+    fn is_verified_final<'db>(&'db self, _db: &'db dyn Database, _input: Id) -> bool {
+        false
```

Comment:
What's the reason that `Input`, `Tracked`, etc. return `false`. Aren't they always considered final?

Reply:
Yeah, good point, I should make these return `true` instead. It doesn't really matter because in practice `is_verified_final` will never be called on any of these, but since we have to put the method on the `Ingredient` trait, it should return sensible values for all ingredient types.